File: Sunrin-ICPC-2022/Final-G/FG.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().split('\n');

let temp = input[0].split(' '), line = 1;
const n = +temp[0], m = +temp[1], q = +temp[2];

let a = new Array(n+2).fill(0);
let x = new Array(n+2).fill(0);
let w = [];
for(let i=0; i<=m; i++) w.push(new Array(n+2).fill(0));

for(let i=1; i<=m; i++){
    const tmp = input[line++].split(' ');
    const cnt = +tmp[0];
    let idx = [];
    for(let j=0; j<cnt; j++) idx.push(+tmp[j+1]);
    idx.push(n+1);
    for(let j=0; j<idx.length; j++) w[i][idx[j]] = +tmp[cnt+j+1];
}

temp = input[line++].split(' ');
for(let i=1; i<=m; i++){
    let t = +temp[i-1];
    for(let j=1; j<=n+1; j++) a[j] += t * w[i][j];
}
a[n+1] += +temp[m];

for(let iter=0; iter<q; iter++){
    const tmp = input[line++].split(' ');
    for(let i=1; i<=n; i++) x[i] = +tmp[i-1];
    x[n+1] = 1;
    res = 0;
    for(let i=1; i<=n+1; i++) res += a[i] * x[i];
    console.log(res);
}

File: Sunrin-ICPC-2022/Final-G/FG.py
n, m, q = map(int, input().split())
a = [0] * (n+2)
x = [0] * (n+2)
w = [[0 for c in range(n+2)] for r in range(m+1)]

for i in range(1, m+1):
    tmp = list(map(int, input().split()))
    cnt = tmp[0]
    idx = []
    for j in range(cnt): idx.append(tmp[j+1])
    idx.append(n+1)
    for j in range(len(idx)):
        w[i][idx[j]] = tmp[cnt+1+j]

tmp = list(map(int, input().split()))
for i in range(1, m+1):
    t = tmp[i-1]
    for j in range(1, n+2): a[j] += t * w[i][j]
a[n+1] += tmp[m]

for iter in range(q):
    tmp = list(map(int, input().split()))
    for i in range(1, n+1): x[i] = tmp[i-1]
    x[n+1] = 1
    res = 0
    for i in range(1, n+2): res += a[i] * x[i]
    print(res)